compiler/compile/OSRData.cpp

```diff
@@ -581,16 +581,16 @@ void TR_OSRCompilationData::buildDefiningMap(TR::Region &region)
       TR::Block *osrCatchBlock = osrMethodData->getOSRCatchBlock();
       if (osrCatchBlock && osrCatchBlock->getExceptionPredecessors().size() > 0)
          {
-         definingMapAtOSRCatchBlocks[i] = new DefiningMap(DefiningMapComparator(), DefiningMapAllocator(comp->trMemory()->currentStackRegion()));
+         definingMapAtOSRCatchBlocks[i] = new (comp->trMemory()->currentStackRegion()) DefiningMap(DefiningMapComparator(), DefiningMapAllocator(comp->trMemory()->currentStackRegion()));
          osrMethodData->buildDefiningMapForBlock(osrCatchBlock, definingMapAtOSRCatchBlocks[i]);
          }
       else osrCatchBlockRemoved = true;
 
       TR::Block *osrCodeBlock = osrMethodData->getOSRCodeBlock();
       if (osrCodeBlock && osrCodeBlock->getPredecessors().size() > 0)
          {
-         definingMapAtOSRCodeBlocks[i] = new DefiningMap(DefiningMapComparator(), DefiningMapAllocator(comp->trMemory()->currentStackRegion()));
-         definingMapAtPrepareForOSRCalls[i] = new DefiningMap(DefiningMapComparator(), DefiningMapAllocator(comp->trMemory()->currentStackRegion()));
+         definingMapAtOSRCodeBlocks[i] = new (comp->trMemory()->currentStackRegion()) DefiningMap(DefiningMapComparator(), DefiningMapAllocator(comp->trMemory()->currentStackRegion()));
```

Comment:
Should these allocations use the `region` that seems to be passed in and used at 

https://github.com/eclipse/omr/blob/dbcaeb952c4c2c314768acc226332cd875ec6f47/compiler/compile/OSRData.cpp#L600

Reply:
note region is the final region not the temporary region - the maps already allocate their internal selves out of the stack region that is destroyed at the end of map building so I believe the placement new should be out of the stack region not the final region for the final map.